lighthouse-core/gather/gatherers/dobetterweb/optimized-images.js

```diff
@@ -88,24 +93,49 @@ class OptimizedImages extends Gatherer {
 
   /**
    * @param {!Object} driver
+   * @param {string} requestId
+   * @param {string} encoding
+   * @return {!Promise<{encodedSize: number}>}
+   */
+  _getEncodedResponse(driver, requestId, encoding) {
+    const quality = encoding === 'jpeg' ? JPEG_QUALITY : WEBP_QUALITY;
+    const params = {requestId, encoding, quality, sizeOnly: true};
+    return driver.sendCommand('Audits.getEncodedResponse', params);
+  }
+
+  /**
+   * @param {!Object} driver
    * @param {{url: string, isBase64DataUri: boolean, resourceSize: number}} networkRecord
    * @return {!Promise<?{originalSize: number, jpegSize: number, webpSize: number}>}
    */
   calculateImageStats(driver, networkRecord) {
-    if (!networkRecord.isSameOrigin && !networkRecord.isBase64DataUri) {
-      // Processing of cross-origin images is buggy and very slow, disable for now
-      // See https://github.com/GoogleChrome/lighthouse/issues/1853
-      // See https://github.com/GoogleChrome/lighthouse/issues/2146
-      return Promise.resolve(null);
-    }
-
-    return Promise.resolve(networkRecord.url).then(uri => {
-      const script = `(${getOptimizedNumBytes.toString()})(${JSON.stringify(uri)})`;
-      return driver.evaluateAsync(script).then(stats => {
-        if (!stats) {
-          return null;
+    // TODO(phulce): remove this dance of trying _getEncodedResponse with a fallback when Audits
+    // domain hits stable in Chrome 62
+    return Promise.resolve(networkRecord.requestId).then(requestId => {
+      if (this._getEncodedResponseUnsupported) return;
+      return this._getEncodedResponse(driver, requestId, 'jpeg').then(jpegData => {
+        return this._getEncodedResponse(driver, requestId, 'webp').then(webpData => {
+          return {
+            originalSize: networkRecord.resourceSize,
+            jpegSize: jpegData.encodedSize,
+            webpSize: webpData.encodedSize,
+          };
+        });
+      }).catch(err => {
+        if (/wasn't found/.test(err.message)) {
+          // Mark non-support so we don't keep attempting the protocol method over and over
+          this._getEncodedResponseUnsupported = true;
```

Comment:
can we include something like `_getEncodedResponseUnsupported` on the `extendedInfo`? Not the most elegant thing, but when looking at old reports (e.g. http archive) it would be good to have an easy way to know when the results include all images, including cross origin ones. Other ways of doing that would also be good (lighthouse version would work in a pinch, but nice to have something audit-local)